staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/fifo_list_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queueset
+
+import (
+	"testing"
+)
+
+func TestFIFO(t *testing.T) {
+	tests := []struct {
+		name   string
+		testFn func(*testing.T, fifo)
+	}{
+		{
+			name: "single request in FIFO, enqueue and then dequeue",
+			testFn: func(t *testing.T, list fifo) {
```

Comment:
nit: I think I'm not a fan of writing the pretty much the whole test body in testFn and bundling them all in a single test.
I think more natural would be to have N tests for them, e.g.:

func TestSingleRequestInFIFO(...) {
  // what you have here
}

func TestXXX(...) {
  ...
}

Reply:
moved to independent tests.